openfeature/multi/multiprovider.go

```diff
@@ -203,53 +213,53 @@ func buildMetadata(m ProviderMap) of.Metadata {
 }
 
 // NewProvider returns a new [multi.Provider] that acts as a unified interface of multiple providers for interaction.
-func NewProvider(providerMap ProviderMap, evaluationStrategy EvaluationStrategy, options ...Option) (*Provider, error) {
-	if len(providerMap) == 0 {
-		return nil, errors.New("providerMap cannot be nil or empty")
-	}
-
+func NewProvider(evaluationStrategy EvaluationStrategy, options ...Option) (*Provider, error) {
 	config := &configuration{
-		logger:        slog.New(slog.DiscardHandler),
-		providerHooks: make(map[string][]of.Hook),
+		logger:    slog.New(slog.DiscardHandler),
+		providers: make([]*namedProvider, 0, 2),
 	}
 
 	for _, opt := range options {
 		opt(config)
 	}
 
-	providers := providerMap
-	collectedHooks := make([]of.Hook, 0, len(providerMap))
-	for name, provider := range providerMap {
+	if len(config.providers) == 0 {
+		return nil, errors.New("no providers configured: at least one provider must be registered using WithProvider()")
+	}
+
+	providers := make([]NamedProvider, 0, len(config.providers))
+	collectedHooks := make([]of.Hook, 0, len(config.providers))
+	for i, provider := range config.providers {
 		// Validate Providers
-		if name == "" {
-			return nil, errors.New("provider name cannot be the empty string")
+		if provider.FeatureProvider == nil {
+			return nil, fmt.Errorf("provider %s at %d cannot be nil", provider.name, i)
 		}
-
-		if provider == nil {
-			return nil, fmt.Errorf("provider %s cannot be nil", name)
+		if provider.name == "" {
+			return nil, fmt.Errorf("provider name at %d cannot be the empty string", i)
 		}
 
 		// Wrap any providers that include hooks
-		if (len(provider.Hooks()) + len(config.providerHooks[name])) == 0 {
+		if (len(provider.Hooks()) + len(provider.extraHooks)) == 0 {
+			providers = append(providers, provider)
 			continue
 		}
 
-		var wrappedProvider of.FeatureProvider
-		if _, ok := provider.(of.EventHandler); ok {
-			wrappedProvider = isolateProviderWithEvents(provider, config.providerHooks[name])
+		var wrappedProvider NamedProvider
+		if _, ok := provider.FeatureProvider.(of.EventHandler); ok {
+			wrappedProvider = isolateProviderWithEvents(provider, provider.extraHooks)
 		} else {
-			wrappedProvider = isolateProvider(provider, config.providerHooks[name])
+			wrappedProvider = isolateProvider(provider, provider.extraHooks)
 		}
 
-		providers[name] = wrappedProvider
+		providers = append(providers, wrappedProvider)
 		collectedHooks = append(collectedHooks, wrappedProvider.Hooks()...)
```

Comment:
```suggestion
		collectedHooks = slices.Concat(collectedHooks, wrappedProvider.Hooks())
```
I know this isn't part of your PR, but just something I noticed could use improvement.